cmake/farm_ng_add_library.cmake

```diff
@@ -0,0 +1,80 @@
+macro(farm_ng_add_library target)
+  set(one_value_args NAMESPACE INCLUDE_DIR INCLUDES MAJOR_VERSION MINOR_VERSION PATCH_VERSION)
+  set(multi_value_args SOURCES HEADERS)
+  cmake_parse_arguments(FARM_NG_ARGS ""
+                        "${one_value_args}" "${multi_value_args}" ${ARGN})
+
+  if(NOT DEFINED FARM_NG_ARGS_NAMESPACE)
+    message(FATAL_ERROR "\nPlease specify NAMESPACE in farm_ng_add_library(${target})\n")
+  endif()
+
+  if(NOT DEFINED FARM_NG_ARGS_INCLUDE_DIR)
+    message(FATAL_ERROR "\nPlease specify INCLUDE_DIR in farm_ng_add_library(${target})\n")
+  endif()
+
+
+  if((NOT DEFINED FARM_NG_ARGS_SOURCES) AND (DEFINED FARM_NG_ARGS_HEADERS))
+    set(INTERFACE_OR_PUBLIC INTERFACE)
```

Comment:
@ethanrublee can we just pass INTERFACE vs. PRIVATE vs. PUBLIC in as an argument? Less magic, and folks google for this and look up what it means.

Reply:
Or we could have:

farm_ng_add_interface

farm_ng_add_pub_library
etc.

Reply:
I'm okay to leave this for a follow-up PR though...